app/scripts/controllers/preferences-controller.ts

```diff
@@ -498,11 +594,14 @@ export default class PreferencesController {
    * Updates the `preferences` property, which is an object. These are user-controlled features
    * found in the settings page.
    *
-   * @param {string} preference - The preference to enable or disable.
-   * @param {boolean |object} value - Indicates whether or not the preference should be enabled or disabled.
-   * @returns {Promise<object>} Promises a new object; the updated preferences object.
+   * @param preference - The preference to enable or disable.
+   * @param value - Indicates whether or not the preference should be enabled or disabled.
+   * @returns Promises a updated Preferences object.
    */
-  async setPreference(preference, value) {
+  async setPreference(
```

Comment:
I believe this function shouldn't be async in first place, once removed the return type would be Preferences